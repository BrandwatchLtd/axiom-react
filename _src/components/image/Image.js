import React, { Component } from 'react';
import { enhance, addPropTypes, addClassName } from '../_utils/components';

export class Image extends Component {
  static propTypes = {
    src: { string: true, isRequired: true },
  };

  render() {
    return (
      <img { ...this.props } />
    );
  }
}

export default enhance(Image)(
  addPropTypes('global'),
  addClassName('global'),
);
