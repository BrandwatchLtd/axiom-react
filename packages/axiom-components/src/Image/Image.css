.ax-image {
  display: block;
  margin-right: auto;
  margin-left: auto;
  border: 0 solid var(--color-theme-border);
}

.ax-image--circle  { border-radius: 50%; }
.ax-image--rounded { border-radius: var(--component-border-radius); }
.ax-image--square  { border-radius: 0; }

.ax-image--border-small { border-width: var(--component-border-width-small); }
.ax-image--border-large { border-width: var(--component-border-width-large); }

.ax-image--border-color-highlight { border-color: var(--color-ui-highlight); }
.ax-image--border-color-success { border-color: var(--color-ui-success); }
.ax-image--border-color-error { border-color: var(--color-ui-error); }
.ax-image--border-color-tiny-clanger { border-color: var(--color-product-tiny-clanger); }
.ax-image--border-color-critical-mass { border-color: var(--color-product-critical-mass); }
.ax-image--border-color-fantastic-voyage { border-color: var(--color-product-fantastic-voyage); }
.ax-image--border-color-paradise-lost { border-color: var(--color-product-paradise-lost); }
.ax-image--border-color-serene-sea { border-color: var(--color-product-serene-sea); }
.ax-image--border-color-electric-dreams { border-color: var(--color-product-electric-dreams); }
.ax-image--border-color-giant-leap { border-color: var(--color-product-giant-leap); }
.ax-image--border-color-moon-lagoon { border-color: var(--color-product-moon-lagoon); }
.ax-image--border-color-space-invader { border-color: var(--color-product-space-invader); }
.ax-image--border-color-terra-form { border-color: var(--color-product-terra-form); }
.ax-image--border-color-primeval-soup { border-color: var(--color-product-primeval-soup); }
.ax-image--border-color-sun-maker { border-color: var(--color-product-sun-maker); }
.ax-image--border-color-new-horizon { border-color: var(--color-product-new-horizon); }
.ax-image--border-color-blast-off { border-color: var(--color-product-blast-off); }
.ax-image--border-color-crash-course { border-color: var(--color-product-crash-course); }
.ax-image--border-color-ground-control { border-color: var(--color-product-ground-control); }
.ax-image--border-color-space-oddity { border-color: var(--color-product-space-oddity); }
.ax-image--border-color-deep-thought { border-color: var(--color-product-deep-thought); }
.ax-image--border-color-luna-dust { border-color: var(--color-product-luna-dust); }
.ax-image--border-color-rgb-ui-carbon { border-color: var(--rgb-ui-carbon); }
