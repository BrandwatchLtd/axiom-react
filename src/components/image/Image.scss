.ax-image {
  display: block;
  max-width: 100%;
}
