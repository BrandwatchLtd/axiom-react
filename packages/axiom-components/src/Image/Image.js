import PropTypes from 'prop-types';
import React, { Component } from 'react';
import classnames from 'classnames';
import Base from '../Base/Base';
import ImageFallback from './ImageFallback';
import './Image.css';

export default class Image extends Component {
  static propTypes = {
    /** Border size around the image */
    border: PropTypes.oneOf(['small', 'large']),
    /** Border colour around the image */
    borderColor: PropTypes.oneOf([
      'highlight',
      'success',
      'error',
      'tiny-clanger',
      'critical-mass',
      'fantastic-voyage',
      'paradise-lost',
      'serene-sea',
      'electric-dreams',
      'giant-leap',
      'moon-lagoon',
      'space-invader',
      'terra-form',
      'primeval-soup',
      'sun-maker',
      'new-horizon',
      'blast-off',
      'crash-course',
      'ground-control',
      'space-oddity',
      'deep-thought',
      'luna-dust',
      'carbon',
    ]),
    /** Fallback content when the image fails to load */
    children: PropTypes.node,
    /** Height of the image. When shape is circle this will be ignored and width will be applied. */
    height: PropTypes.string,
    /** @type {[type]} [description] */
    maxWidth: PropTypes.string,
    /** SKIP */
    onError: PropTypes.func,
    /** SKIP */
    onLoad: PropTypes.func,
    /** Shape of the image */
    shape: PropTypes.oneOf(['circle', 'rounded', 'square']),
    /** Source of the image */
    src: PropTypes.string,
    /** Width of the image. When shape is circle this will also be used for height. */
    width: PropTypes.string,
  };

  static defaultProps = {
    maxWidth: '100%',
    shape: 'square',
  };

  render() {
    const {
      border,
      borderColor,
      children,
      height,
      maxWidth,
      onError,
      onLoad,
      shape,
      src,
      width,
      ...rest
    } = this.props;

    const style = {
      maxWidth,
      width: shape === 'circle' ? (width || height) : width,
      height: shape === 'circle' ? (width || height) : height,
    };

    const classes = classnames('ax-image', `ax-image--${shape}`, {
      [`ax-image--border-${border}`]: border,
      [`ax-image--border-color-${borderColor}`]: borderColor,
    });

    return (
      <ImageFallback
          fallback={ children }
          onError={ onError }
          onLoad={ onLoad }
          src={ src }>
        <Base
            space="x4"
            { ...rest }
            Component="img"
            className={ classes }
            src={ src }
            style={ style } />
      </ImageFallback>
    );
  }
}
