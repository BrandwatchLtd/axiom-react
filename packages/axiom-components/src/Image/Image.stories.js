import React from "react";
import Image from "./Image";
import AlertBar from "../AlertBar/AlertBar";

export default {
  title: "Components/Image",
  component: Image,
};

export function Default() {
  return (
    <Image src="/avatar.png">
      <AlertBar type="error">Image failed to load</AlertBar>
    </Image>
  );
}

export function WithFallback() {
  return (
    <Image src="badPath">
      <AlertBar type="error">Image failed to load</AlertBar>
    </Image>
  );
}

export function Shape() {
  return (
    <div>
      <Image shape="square" src="/avatar.png" width="50px" />
      <Image shape="circle" src="/avatar.png" width="50px" />
      <Image shape="rounded" src="/avatar.png" width="50px" />
    </div>
  );
}
