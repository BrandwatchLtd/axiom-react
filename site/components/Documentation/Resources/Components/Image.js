import React, { Component } from 'react';
import { AlertBar, Image } from '@brandwatch/axiom-components';
import {
  DocumentationApi,
  DocumentationContent,
  DocumentationShowCase,
} from '@brandwatch/axiom-documentation-viewer';

export default class Documentation extends Component {
  render() {
    return (
      <DocumentationContent>
        <DocumentationShowCase>
          <Image src="/assets/axiom-bg.jpg">
            <AlertBar type="error">Image failed to load</AlertBar>
          </Image>
        </DocumentationShowCase>

        <DocumentationApi
          components={[
            require('!!axiom-documentation-loader!@brandwatch/axiom-components/src/Image/Image'),
            require('!!axiom-documentation-loader!@brandwatch/axiom-components/src/Image/ImageFallback'),
          ]}
        />
      </DocumentationContent>
    );
  }
}
