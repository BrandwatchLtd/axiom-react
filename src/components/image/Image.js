import React, { Component, PropTypes } from 'react';
import { Base } from 'bw-axiom';

if (__INCLUDE_CSS__) {
  require('../image/Image.scss');
}

export default class Image extends Component {
  static propTypes = {
    children: PropTypes.node,
  };

  constructor(props) {
    super(props);

    this.state = {
      fallback: false,
    };
  }

  showFallback() {
    this.setState({ fallback: true });
  }

  render() {
    const { children, ...rest } = this.props;
    const { fallback } = this.state;

    if (fallback) {
      return children;
    }

    return (
      <Base Component="img" { ...rest } onError={ children && ::this.showFallback } />
    );
  }
}
