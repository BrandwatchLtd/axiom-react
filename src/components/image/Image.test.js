import React from 'react';
import renderer from 'react-test-renderer';
import Image from './Image';

const getComponent = (props = {}) =>
  renderer.create(
    <Image src="/image.png" { ...props } />
  );

describe('Image', () => {
  it('renders with defaultProps', () => {
    const component = getComponent();
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with extra props', () => {
    const component = getComponent({ title: 'Lorem', alt: 'Lorem' });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
