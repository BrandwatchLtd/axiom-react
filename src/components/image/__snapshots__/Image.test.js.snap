exports[`Image renders with defaultProps 1`] = `
<img
  className=""
  src="/image.png" />
`;

exports[`Image renders with extra props 1`] = `
<img
  alt="Lorem"
  className=""
  src="/image.png"
  title="Lorem" />
`;
